Config/Controller/ControllerConfig.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the `liip/LiipImagineBundle` project.
+ *
+ * (c) https://github.com/liip/LiipImagineBundle/graphs/contributors
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Liip\ImagineBundle\Config\Controller;
+
+use Liip\ImagineBundle\Exception\InvalidArgumentException;
+
+final class ControllerConfig
+{
+    /**
+     * @var int
+     */
+    private $redirectResponseCode;
+
+    /**
+     * @param int $redirectResponseCode
+     */
+    public function __construct(int $redirectResponseCode)
+    {
+        if (!in_array($redirectResponseCode, [201, 301, 302, 303, 307, 308], true)) {
```

Comment:
i would prefer to put this validation into the configuration definition / DI class.

that way you get an exception right away when building the container, and not only when actually using the controller.

Reply:
yeah, it makes perfect sense in the configuration definition, but would you leave also this validation here? I think it should also belong here

Reply:
yep, it makes sense to validate here as people could be configuring the service themselves. how about defining a public constant with this array in this class and use that constant in the validation of Configuration.php ?